src/realm/obj.cpp

```diff
@@ -222,6 +222,37 @@ CollectionListPtr CollectionList::get_collection_list(size_t ndx) const
     return CollectionList::create(const_cast<CollectionList*>(this)->shared_from_this(), m_col_key, index, coll_type);
 }
 
+void CollectionList::remove(size_t ndx)
+{
+    REALM_ASSERT(m_key_type == type_Int);
+    auto int_keys = static_cast<BPlusTree<Int>*>(m_keys.get());
+    if (ndx < int_keys->size()) {
+        int_keys->erase(ndx);
```

Comment:
I would prefer to invert the check and throw here.

Reply:
ok, done